### ISSUE TITLE
"could not set signature algorithms" when running Servo on Android

### ISSUE THREAD

**Issue by zhuowei** [servo/servo#25619]
When I launch Servo on Android, nothing happens: the page stays blank, and trying to navigate to any site does nothing.

In the logcat, there are these two error messages:

```
01-26 17:20:56.332  5655  5704 W simpleservo: constellation::constellation: Visibility change for closed browsing context PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }.
01-26 17:20:56.416  5655  5683 D simpleservo: thread 'ResourceManager' panicked at 'could not set signature algorithms: ErrorStack([])', components/net/connector.rs:97:5
01-26 17:20:56.416  5655  5683 D simpleservo: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

```
01-26 17:22:30.358  5655  5680 I simpleservo: simpleservo: load_uri: https://example.org/
01-26 17:22:30.359  5655  5704 W simpleservo: constellation::constellation: LoadUrl for unknow browsing context: TopLevelBrowsingContextId(BrowsingContextId { namespace_id: PipelineNamespaceId(0), index: BrowsingContextIndex(1) })
```

Android 10, Pixel 3 XL, Servo nightly build downloaded January 26th.

The call to `set_sigalgs_list` was added in #24976. Does the version of OpenSSL 1.0.2k used o n Android actually support all the signature algorithms specified there? The source doesn't mention ed448 at all:

```
:~/servo-work/servo/target/android/aarch64-linux-android/native/openssl/openssl-1.0.2k$ grep -r ed448
:~/servo-work/servo/target/android/aarch64-linux-android/native/openssl/openssl-1.0.2k$ 
```

**Comment by Darkspirit** [comment]
I guess your analysis is correct. set_sigalgs_list() can be used with [OpenSSL 1.0.2](https://github.com/sfackler/rust-openssl/blob/ad37e7e07db148e7f0114a2f44c1196a9d35d3c3/openssl/src/ssl/mod.rs#L1705) and later. I expected that unknown signature algorithms would be ignored. The best fix is upgrading OpenSSL to 1.1.1 for Android as well. That would also bring TLS 1.3 support to this target.

**Comment by Darkspirit** [comment]
We need https://www.openssl.org/source/openssl-1.1.1d.tar.gz at this location:
https://github.com/servo/servo/blob/7596a0fc1a6cd5880c12d3a785236ea4d71f0be7/support/android/openssl.makefile#L13

These should be changed to 1.1.1d:
https://github.com/servo/servo/blob/ce6204b904e62307915e0870611bad473cd48843/python/servo/build_commands.py#L334
https://github.com/servo/servo/blob/ce6204b904e62307915e0870611bad473cd48843/python/servo/build_commands.py#L583

[openssl.sh](https://github.com/servo/servo/blob/7596a0fc1a6cd5880c12d3a785236ea4d71f0be7/support/android/openssl.sh) might need a small change: https://wiki.openssl.org/index.php/Android
> NOTE: The instructions on this page are for older versions of OpenSSL (they should work for OpenSSL 1.0.2). For OpenSSL 1.1.1 see the [INSTALL](https://github.com/openssl/openssl/blob/master/INSTALL) and [NOTES.ANDROID](https://github.com/openssl/openssl/blob/master/NOTES.ANDROID) files in the source distribution.
